// Code generated by mockery v2.10.0. DO NOT EDIT.

package mocks

import (
	usecase "github.com/mkaiho/go-lambda-api-sample/usecase"
	mock "github.com/stretchr/testify/mock"
)

// UseCaseError is an autogenerated mock type for the UseCaseError type
type UseCaseError struct {
	mock.Mock
}

// Error provides a mock function with given fields:
func (_m *UseCaseError) Error() string {
	ret := _m.Called()

	var r0 string
	if rf, ok := ret.Get(0).(func() string); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(string)
	}

	return r0
}

// Type provides a mock function with given fields:
func (_m *UseCaseError) Type() usecase.ErrorType {
	ret := _m.Called()

	var r0 usecase.ErrorType
	if rf, ok := ret.Get(0).(func() usecase.ErrorType); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(usecase.ErrorType)
	}

	return r0
}
